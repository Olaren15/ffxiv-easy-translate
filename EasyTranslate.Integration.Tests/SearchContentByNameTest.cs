using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;
using System.Linq;
using System.Threading.Tasks;
using EasyTranslate.Domain.Entities;
using EasyTranslate.UseCase;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using Xunit;

namespace EasyTranslate.Integration.Tests;

// Assertions are done with snapshots
[SuppressMessage("Blocker Code Smell", "S2699:Tests should include assertions")]
public class SearchContentByNameTest
{
    private readonly SearchContentByNameUseCase _searchContentByNameUseCase;

    public SearchContentByNameTest()
    {
        IServiceProvider serviceProvider = ServiceProviderBuilder.Build();
        _searchContentByNameUseCase = serviceProvider.GetService<SearchContentByNameUseCase>()!;
    }

    [Fact]
    public async Task Item_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Popoto",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Action_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "cure",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }


    [Fact]
    public async Task Achievement_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "For the hoard",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task BNpcname_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "bhoot",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task ClassJob_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute("summoner", Language.English);

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Companion_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "lesser panda",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task ContentFinderCondition_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "sastasha",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task CraftAction_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Groundwork",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Emote_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "pet",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task ENpcResident_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "G'raha Tia",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task EventItemLite_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Ulan's note",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Fate_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Drink Me",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Leve_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Practical Command",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Mount_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Demi-Ozma",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Orchestrion_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "pa-paya",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task PlaceName_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Limsa Lominsa",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Quest_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "For Want Of a Memory",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Status_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Vulnerability Up",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Title_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Goddess of Magic",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Trait_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Maim and Mend",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Weather_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Fair Skies",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }
}
