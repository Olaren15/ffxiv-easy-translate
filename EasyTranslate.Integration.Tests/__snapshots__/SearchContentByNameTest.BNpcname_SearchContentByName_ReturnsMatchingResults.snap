[
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "bhoot",
    "FrenchName": "bhut",
    "GermanName": "Bhut",
    "JapaneseName": "ブフート"
  },
  {
    "Type": "Fate",
    "IconId": 60721,
    "EnglishName": "Das Bhoot",
    "FrenchName": "Le don du repos éternel",
    "GermanName": "Und was sich reimt, ist Bhut",
    "JapaneseName": "エクソシスト"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "polis bhoot",
    "FrenchName": "bhut de l'Aquapole",
    "GermanName": "Aquapolis-Bhut",
    "JapaneseName": "アクアポリス・ブフート"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "gelid bhoot",
    "FrenchName": "bhut glacial",
    "GermanName": "eisig[a] Bhut",
    "JapaneseName": "ジェリド・ブフート"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "guard bhoot",
    "FrenchName": "bhut gardien",
    "GermanName": "Wächter-Bhut",
    "JapaneseName": "ガード・ブフート"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "Pyros bhoot",
    "FrenchName": "bhut Pyros",
    "GermanName": "Pyros-Bhut",
    "JapaneseName": "ピューロス・ブフート"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "Mhachi bhoot",
    "FrenchName": "bhut mhachois",
    "GermanName": "Mhach-Bhut",
    "JapaneseName": "マハ・ブフート"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "broken bhoot",
    "FrenchName": "bhut écorché",
    "GermanName": "Klippen-Bhut",
    "JapaneseName": "ブロークン・ブフート"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "impure bhoot",
    "FrenchName": "bhut impur",
    "GermanName": "unrein[a] Bhut",
    "JapaneseName": "インピュア・ブフート"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "Zadnor bhoot",
    "FrenchName": "bhut de Zadnor",
    "GermanName": "Zadnor-Bhut",
    "JapaneseName": "ザトゥノル・ブフート"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "Orthos bhoot",
    "FrenchName": "bhut Orthos",
    "GermanName": "Orthos-Bhut",
    "JapaneseName": "オルト・ブフート"
  },
  {
    "Type": "Quest",
    "IconId": 71221,
    "EnglishName": " Bhoot Camps",
    "FrenchName": " Les anges de l'insalubrité",
    "GermanName": " Geisterstunde",
    "JapaneseName": " 浮遊する布"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "horrific bhoot",
    "FrenchName": "bhut horrifique",
    "GermanName": "schrecklich[a] Bhut",
    "JapaneseName": "ホリフィック・ブフート"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "duskfall bhoot",
    "FrenchName": "bhut nocturne",
    "GermanName": "Dämmer-Bhut",
    "JapaneseName": "ナイト・ブフート"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "nightmare bhoot",
    "FrenchName": "bhut du cauchemar",
    "GermanName": "Albtraum-Bhut",
    "JapaneseName": "ナイトメア・ブフート"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "deep palace bhoot",
    "FrenchName": "bhut des profondeurs",
    "GermanName": "Katakomben-Bhut",
    "JapaneseName": "ディープパレス・ブフート"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "somnambulant bhoot",
    "FrenchName": "bhut somnambule",
    "GermanName": "Schlafwandler-Bhut",
    "JapaneseName": "ソムナムブラント・ブフート"
  }
]
